test_conformance/api/test_api_consistency.cpp

```diff
@@ -0,0 +1,479 @@
+        test_error(error, "Unable to create test kernel");
+
+        // clGetMemObjectInfo, passing CL_MEM_USES_SVM_POINTER
+        // Returns CL_FALSE if no devices in the context associated with
+        // memobj support Shared Virtual Memory.
+        cl_bool usesSVMPointer;
+        error =
+            clGetMemObjectInfo(mem, CL_MEM_USES_SVM_POINTER,
+                               sizeof(usesSVMPointer), &usesSVMPointer, NULL);
+        test_error(error, "Unable to query CL_MEM_USES_SVM_POINTER");
+        if (usesSVMPointer != CL_FALSE)
+        {
+            log_error("CL_DEVICE_SVM_CAPABILITIES returned 0 but "
+                      "CL_MEM_USES_SVM_POINTER did not return CL_FALSE\n");
+            return TEST_FAIL;
+        }
+
+        // Check that the SVM APIs can be called.
+
+        // Returns NULL if no devices in context support Shared Virtual Memory.
+        void* ptr0 = clSVMAlloc(context, CL_MEM_READ_WRITE, allocSize, 0);
+        void* ptr1 = clSVMAlloc(context, CL_MEM_READ_WRITE, allocSize, 0);
+        if (ptr0 != NULL || ptr1 != NULL)
+        {
+            log_error("CL_DEVICE_SVM_CAPABILITIES returned 0 but clSVMAlloc "
+                      "returned a non-NULL value\n");
+            return TEST_FAIL;
+        }
+
+        // clEnqueueSVMFree, clEnqueueSVMMemcpy, clEnqueueSVMMemFill,
+        // clEnqueueSVMMap, clEnqueueSVMUnmap, clEnqueueSVMMigrateMem Returns
+        // CL_INVALID_OPERATION if the device associated with command_queue does
+        // not support Shared Virtual Memory.
+
+        cl_uint pattern = 0xAAAAAAAA;
+        error = clEnqueueSVMMemFill(queue, ptr0, &pattern, sizeof(pattern),
+                                    allocSize, 0, NULL, NULL);
+        test_failure_error(
+            error, CL_INVALID_OPERATION,
+            "CL_DEVICE_SVM_CAPABILITIES returned 0 but clEnqueueSVMMemFill did "
+            "not return CL_INVALID_OPERATION");
+
+        error = clEnqueueSVMMemcpy(queue, CL_TRUE, ptr1, ptr0, allocSize, 0,
+                                   NULL, NULL);
+        test_failure_error(
+            error, CL_INVALID_OPERATION,
+            "CL_DEVICE_SVM_CAPABILITIES returned 0 but "
+            "clEnqueueSVMMemcpy did not return CL_INVALID_OPERATION");
+
+        error = clEnqueueSVMMap(queue, CL_TRUE, CL_MAP_READ, ptr1, allocSize, 0,
+                                NULL, NULL);
+        test_failure_error(
+            error, CL_INVALID_OPERATION,
+            "CL_DEVICE_SVM_CAPABILITIES returned 0 but "
+            "clEnqueueSVMMap did not return CL_INVALID_OPERATION");
+
+        error = clEnqueueSVMUnmap(queue, ptr1, 0, NULL, NULL);
+        test_failure_error(
+            error, CL_INVALID_OPERATION,
+            "CL_DEVICE_SVM_CAPABILITIES returned 0 but "
+            "clEnqueueSVMUnmap did not return CL_INVALID_OPERATION");
+
+        error = clFinish(queue);
+        test_error(error, "Error calling clFinish after SVM operations");
```

Comment:
Why call `clFinish` here? The test will have bailed out if any enqueue command succeeded.

Reply:
Yes, I suppose this isn't required since we're checking and enforcing that all of the preceding commands return errors, but it would be required otherwise to ensure the SVM operations are complete before freeing the pointers.  Since it's not hurting anything I'm inclined to keep it, but I don't care strongly either way.

Reply:
I think this is confusing and we should remove it. We don't want the test to accidentally work on a broken implementation because of this.

Reply:
I'll replace these calls with comments instead.